filebeat/fileset/config.go

```diff
@@ -0,0 +1,17 @@
+package fileset
+
+// ModuleConfig contains the configuration file options for a module
+type ModuleConfig struct {
+	Module   string                    `config:"module"     validate:"required"`
+	Enabled  *bool                     `config:"enabled"`
+	Filesets map[string]*FilesetConfig `config:"filesets"`
```

Comment:
The reason for prefixing with `filesets` is just to keep the implementation simple, in the sense that the ModuleConfig can be just:

```
type ModuleConfig struct {
	Module   string                    `config:"module"     validate:"required"`
	Enabled  *bool                     `config:"enabled"`
	Filesets map[string]*FilesetConfig `config:"filesets"`
}
```

And future module level settings can be added easily. In order to get rid of `filesets`, I'd need something like:

```
type ModuleConfig map[string]*common.Config
```

and then have code to extract the "enabled" and "name". I can look into doing that, although I'm not sure the extra code complexity is justified.

Btw, I wrote an overview on the user interaction here: https://github.com/elastic/beats/issues/3159